### ISSUE TITLE
rotors_gazebo_plugins:make error

### ISSUE THREAD

**Issue by NoemiF99** [gsilano/CrazyS#105]
![Immagine 2023-09-22 183633](https://github.com/gsilano/CrazyS/assets/145793961/b4664817-2318-48e8-990e-7dfcafed19cc)
![Immagine 2023-09-22 183710](https://github.com/gsilano/CrazyS/assets/145793961/98d51a70-a515-411f-9997-03017fa41942)
![Immagine 2023-09-22 183752](https://github.com/gsilano/CrazyS/assets/145793961/b2808782-4de8-46bb-bb71-565c88642874)
How can I solve this problem?

**Comment by gsilano** [comment]
Hello! It's a bit unclear to me what's happening in that context. Could you please provide more information regarding the Ubuntu and ROS releases you're referring to? Thank you!

Have you had the opportunity to review the resolved issues? The problem may have already been resolved.

**Comment by NoemiF99** [comment]
hello, I have ubunto 20.04.2.0, ros noetic e Gazebo multi-robot simulator, version 11.11.0 with Virtual Box. Installed CrazyS following ReadMe.md and when i run 'catkin build', I encounter the issues i shared in my previous post. I read solution #590, which had the same error as mine, and it said to follow the ReadMe.md, which I followed to use CrazyS, but every time I try to 'catkin build,' it gives me an error in the build of rotors_gazebo_plugins

**Comment by gsilano** [comment]
I see.  It appears that some updates may be missing. Please rerun the following commands and give me back with the results.

```
$ sudo sh -c 'echo "deb http://packages.ros.org/ros/ubuntu $(lsb_release -sc) main" > /etc/apt/sources.list.d/ros-latest.list'
$ sudo apt install curl # if you haven't already installed curl
$ curl -s https://raw.githubusercontent.com/ros/rosdistro/master/ros.asc | sudo apt-key add -
$ sudo apt update
$ sudo apt install ros-noetic-desktop-full ros-noetic-joy ros-noetic-octomap-ros ros-noetic-mavlink
$ sudo apt install ros-noetic-octomap-mapping ros-noetic-control-toolbox
$ sudo apt install python3-vcstool python3-catkin-tools protobuf-compiler libgoogle-glog-dev
$ sudo rosdep init
$ rosdep update
$ echo "source /opt/ros/noetic/setup.bash" >> ~/.bashrc
$ source ~/.bashrc
$ sudo apt-get install python3-rosdep python3-wstool ros-noetic-ros libgoogle-glog-dev
$ rosdep install --from-paths src -i
$ rosdep update
$ catkin config --cmake-args -DCMAKE_BUILD_TYPE=Release -DCATKIN_ENABLE_TESTING=False
$ catkin build
```

**Comment by RugvedKatole** [comment]
Hi, I am facing the same error. I ran all the commands you provided here. 
I have attached the output given by the catkin build command.
[errors.txt](https://github.com/gsilano/CrazyS/files/14875405/errors.txt)

**Comment by gsilano** [comment]
Hi there! Have you tried following the commands mentioned above?

**Comment by RugvedKatole** [comment]
Yes I tried following the commands mentioned above. The build error thrown by catkin build command is store inside error.txt file above.

**Comment by gsilano** [comment]
@RugvedKatole , read #40 and let me know if this solves the issue.

**Comment by Faivex** [comment]
Hi all,
I had almost the same problem.
I've been trying to build CrazyS on Windows Subsystem for Linux (WSL), using ```Ubuntu 20.04```, ROS Noetic, and ```Gazebo 11```.
Every time I tried to build ```rotors_gazebo``` and ```rotors_simulator``` an error was raised.
As a result, when I launched the hovering example, the drone didn't fly.
After testing many possible solutions, I discovered it is because the swap is 1GB on my WSL (you can find it by ```swapon --show```), which was insufficient. 
I simply increased it to 4GB (see [here](https://www.youtube.com/watch?v=Tu95sdnALJk)) and the problem was solved.
Hope this helps.

**Comment by kashifzr** [comment]
@gsilano i need to know how to use plugins define like librotors_gazebo_lidar_plugins.so define in rotors_gazebo_plugins. I am using lidar in component snippet where different plugins with cpp code are given. When i used plugins given by gazebo like libgazebo_ros_laser.so it give me scan values and it does work as expected but I want to used some customized plugins in my codes.
<pre><font color="#06989A"><b>libLiftDragPlugin.so</b></font>
<font color="#06989A"><b>libroll_pitch_yawrate_thrust_controller.so</b></font>
<font color="#06989A"><b>librotors_gazebo_bag_plugin.so</b></font>
<font color="#06989A"><b>librotors_gazebo_controller_interface.so</b></font>
<font color="#06989A"><b>librotors_gazebo_fw_dynamics_plugin.so</b></font>
<font color="#06989A"><b>librotors_gazebo_gps_plugin.so</b></font>
<font color="#06989A"><b>librotors_gazebo_imu_plugin.so</b></font>
<font color="#06989A"><b>librotors_gazebo_lidar_plugin.so</b></font>
<font color="#06989A"><b>librotors_gazebo_magnetometer_plugin.so</b></font>
<font color="#06989A"><b>librotors_gazebo_motor_model.so</b></font>
<font color="#06989A"><b>librotors_gazebo_multirotor_base_plugin.so</b></font>
<font color="#06989A"><b>librotors_gazebo_noisydepth_plugin.so</b></font>
<font color="#06989A"><b>librotors_gazebo_odometry_plugin.so</b></font>
<font color="#06989A"><b>librotors_gazebo_pressure_plugin.so</b></font>
<font color="#06989A"><b>librotors_gazebo_ros_interface_plugin.so</b></font>
<font color="#06989A"><b>librotors_gazebo_velodyne_plugin.so</b></font>
<font color="#06989A"><b>librotors_gazebo_wind_plugin.so</b></font>
<font color="#3465A4"><b>pkgconfig</b></font>
<font color="#3465A4"><b>python3</b></font>
<font color="#3465A4"><b>rotors_control</b></font>
<font color="#3465A4"><b>rotors_evaluation</b></font>
<font color="#3465A4"><b>rotors_gazebo</b></font>
<font color="#3465A4"><b>rotors_joy_interface</b></font>
</pre>
in component_snippet all these plugins are being referred but librotor_gazebo_lidar_plugin is not used but given in rotors_gazebo_plugins. Hope this make sense

**Comment by gsilano** [comment]
Hi! I think you should use the [issue tracker of the RotorS package](https://github.com/ethz-asl/rotors_simulator) for this type of question.

**Comment by IngCristianVC** [comment]
> Hi all, I had almost the same problem. I've been trying to build CrazyS on Windows Subsystem for Linux (WSL), using `Ubuntu 20.04`, ROS Noetic, and `Gazebo 11`. Every time I tried to build `rotors_gazebo` and `rotors_simulator` an error was raised. As a result, when I launched the hovering example, the drone didn't fly. After testing many possible solutions, I discovered it is because the swap is 1GB on my WSL (you can find it by `swapon --show`), which was insufficient. I simply increased it to 4GB (see [here](https://www.youtube.com/watch?v=Tu95sdnALJk)) and the problem was solved. Hope this helps.

hey, the video says that is private, i had this problem with the catkin build
